kiarash/TS/TS: OOP/src/index.ts

```diff
@@ -0,0 +1,62 @@
+type Branch = "architecture" | "clean-code" | "languages";
+
+abstract class Concept {
+  constructor(protected branch: string, private name: string) {}
+
+  getName() {
+    return this.name;
+  }
+
+  getBranch() {
+    return this.branch;
+  }
+}
+
+class DevConcept extends Concept {
+  constructor(branch: Branch) {
+    super(branch, "DevConcept");
+  }
+}
+
+class PhilosophicalConcept extends Concept {
+  private reasoning_list: string[] = [];
+  constructor(branch: string, reasoning?: string[]) {
+    super(branch, "PhilosophicalConcept");
+    if (reasoning) {
+      this.reasoning_list = [...this.reasoning_list, ...reasoning];
+    }
+  }
+
+  getBranch(): string {
```

Comment:
This new method has an explicit return type, which shouldn't, considering consistency comment.